comp/core/status/statusimpl/common_header_provider.go

```diff
@@ -0,0 +1,106 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package statusimpl
+
+import (
+	"fmt"
+	htmlTemplate "html/template"
+	"io"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	textTemplate "text/template"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/status"
+	"github.com/DataDog/datadog-agent/pkg/collector/python"
+	pkgConfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+var nowFunc = time.Now
+var startTimeProvider = pkgConfig.StartTime
+
+type headerProvider struct {
+	data                   map[string]interface{}
+	name                   string
+	textTemplatesFunctions textTemplate.FuncMap
+	htmlTemplatesFunctions htmlTemplate.FuncMap
+}
+
+func (h *headerProvider) Index() int {
+	return 0
+}
+
+func (h *headerProvider) Name() string {
+	return h.name
+}
+
+func (h *headerProvider) JSON(stats map[string]interface{}) {
+	for k, v := range h.data {
+		stats[k] = v
+	}
+}
+
+func (h *headerProvider) Text(buffer io.Writer) error {
+	tmpl, tmplErr := templatesFS.ReadFile(path.Join("templates", "text.tmpl"))
+	if tmplErr != nil {
+		return tmplErr
+	}
+	t := textTemplate.Must(textTemplate.New("header").Funcs(h.textTemplatesFunctions).Parse(string(tmpl)))
+	return t.Execute(buffer, h.data)
+}
+
+func (h *headerProvider) HTML(buffer io.Writer) error {
+	tmpl, tmplErr := templatesFS.ReadFile(path.Join("templates", "html.tmpl"))
+	if tmplErr != nil {
+		return tmplErr
+	}
+	t := htmlTemplate.Must(htmlTemplate.New("header").Funcs(h.htmlTemplatesFunctions).Parse(string(tmpl)))
+	return t.Execute(buffer, h.data)
+}
+
+func newCommonHeaderProvider(config config.Component) status.HeaderProvider {
+	configObject := config.Object()
```

Comment:
This is not needed, the component act like a `Reader` (see [this](https://github.com/DataDog/datadog-agent/blob/main/comp/core/config/component.go#L32)).